mace/modules/models.py

```diff
@@ -254,120 +187,364 @@ def forward(
-            )
-            node_es_list.append(readout(node_feats).squeeze(-1))  # {[n_nodes, ], }
 
-        # Sum over interactions
-        node_inter_es = torch.sum(
-            torch.stack(node_es_list, dim=0), dim=0
-        )  # [n_nodes, ]
-        node_inter_es = self.scale_shift(node_inter_es)
+@compile_mode("script")
+class EnergyDipolesMACE(MACE, DipoleModelMixin):
+    """MACE model for Energy & Dipoles"""
+
+    _LINEAR_READOUT_CLASS = LinearDipoleReadoutBlock
+    _NONLINEAR_READOUT_CLASS = NonLinearDipoleReadoutBlock
+
+    def __init__(self, hidden_irreps: o3.Irreps, **kwargs):
+        """MACE energy & dipole model
+
+        Parameters
+        ----------
+        # ------- core model parameters
+        r_max
+            cutoff of radial embedding applied to individual atoms
+        num_bessel
+            number of Bessel functions to be used for radial embedding
+        num_polynomial_cutoff
+        max_ell
+            l_max, maximum channels for spherical harmonics
+        interaction_cls
+            class to be used for interactions blocks
+        interaction_cls_first
+            class to be used for the first layer's interaction block
+        num_interactions
+            number of interaction layers to use
+        num_elements
+            redundant parameter for the number of elements
+        hidden_irreps
+            hidden irreducible representations, basically the size of the layer features
+            and hence direct control on the size of the model
+        MLP_irreps
+        avg_num_neighbors
+        atomic_numbers
+            atomic numbers of elements the model supports
+            order & length to agree with atomic_energies
+        correlation
+        gate
+            non-linearity for non-linear readouts
+        radial_MLP
+
+        # ------- energy model specific parameters
+        atomic_energies
+            energy shift (e0) of elements
+            order & length to agree with atomic_numbers
+
+        **kwargs
+
+        Notes
+        -----
+        Requires at least l=1 features to be used for representing dipoles.
+        `atomic_energies` parameter added for compatibility, but it's ignored
+
+        """
+        assert (
+            hidden_irreps.lmax >= 1
+        ), "To predict dipoles use at least l=1 hidden_irreps"
+        super().__init__(hidden_irreps=hidden_irreps, **kwargs)
+
+    @staticmethod
+    def _last_layer_irreps(hidden_irreps) -> o3.Irreps:
+        """energy & dipole: Select scalars and l=1 vectors for last layer"""
+        return o3.Irreps(str(hidden_irreps[:2]))
 
-        # Sum over nodes in graph
-        inter_e = scatter_sum(
-            src=node_inter_es, index=data["batch"], dim=-1, dim_size=num_graphs
-        )  # [n_graphs,]
+    def forward(
+        self,
+        data: Dict[str, torch.Tensor],
+        training: bool = False,
+        compute_force: bool = True,
```

Comment:
Same here, parameters as traps only. Can understand if it is compatibility and perhaps will be implemented later on.